File: edgequake/sdks/typescript/src/resources/api-keys.ts
/**
 * API Keys resource — create, list, revoke API keys.
 *
 * WHY: Updated to match Rust ListApiKeysResponse and RevokeApiKeyResponse.
 * @module resources/api-keys
 * @see edgequake/crates/edgequake-api/src/handlers/auth.rs
 */

import type {
  ApiKeyResponse,
  CreateApiKeyRequest,
  ListApiKeysResponse,
  RevokeApiKeyResponse,
} from "../types/auth.js";
import { Resource } from "./base.js";

export class ApiKeysResource extends Resource {
  /** Create a new API key. */
  async create(request: CreateApiKeyRequest): Promise<ApiKeyResponse> {
    return this._post("/api/v1/api-keys", request);
  }

  /**
   * List all API keys.
   * WHY: Rust returns ListApiKeysResponse { keys, total, page, page_size, total_pages }.
   */
  async list(query?: {
    page?: number;
    page_size?: number;
  }): Promise<ListApiKeysResponse> {
    const params = new URLSearchParams();
    if (query?.page) params.set("page", String(query.page));
    if (query?.page_size) params.set("page_size", String(query.page_size));
    const qs = params.toString();
    return this._get(`/api/v1/api-keys${qs ? `?${qs}` : ""}`);
  }

  /**
   * Revoke (delete) an API key.
   * WHY: Rust returns RevokeApiKeyResponse { key_id, message }.
   */
  async revoke(keyId: string): Promise<RevokeApiKeyResponse> {
    return this._del(`/api/v1/api-keys/${keyId}`);
  }
}

File: edgequake/sdks/rust/src/resources/api_keys.rs
//! API Keys resource.

use crate::client::EdgeQuakeClient;
use crate::error::Result;
use crate::types::auth::*;

pub struct ApiKeysResource<'a> {
    pub(crate) client: &'a EdgeQuakeClient,
}

impl<'a> ApiKeysResource<'a> {
    /// `GET /api/v1/api-keys`
    pub async fn list(&self) -> Result<Vec<ApiKeyInfo>> {
        self.client.get("/api/v1/api-keys").await
    }

    /// `POST /api/v1/api-keys`
    pub async fn create(&self, name: &str) -> Result<ApiKeyResponse> {
        let body = serde_json::json!({ "name": name });
        self.client.post("/api/v1/api-keys", Some(&body)).await
    }

    /// `DELETE /api/v1/api-keys/{id}`
    pub async fn revoke(&self, id: &str) -> Result<()> {
        self.client
            .delete_no_content(&format!("/api/v1/api-keys/{id}"))
            .await
    }
}
